src/fromager/resolver.py

```diff
@@ -509,3 +509,52 @@ def _find_tags(
                 yield entry["tarball_url"], version
             # pagination links
             nexturl = resp.links.get("next", {}).get("url")
+
+
+class CustomPackageProvider(BaseProvider):
+    """Get best match"""
+
+    def __init__(
+        self,
+        req: Requirement,
+        constraints: Constraints,
+        versions: dict[Version, str],
+    ):
+        super().__init__(constraints=constraints)
+        self.req = req
+        self.versions = versions
+
+    def get_cache(self) -> dict[str, list[Candidate]]:
+        return GenericProvider.generic_resolver_cache
+
+    def find_best_match(
+        self,
+    ) -> Version:
+        # convert self.versions to candidates
+        candidates = []
+        for version, _ in self.versions.items():
```

Comment:
Do the other providers sort the list of versions to evaluate? I would expect the order to matter. Maybe that happens after the candidates are produced, though?

Reply:
Yes, the order in builder/package_plugins is older to newer (ie latest).  This is a standard.  I was thinking that a sort may be necessary but for now left it out.  Either way, this does enforce the older to newer standard in the plugins which we've been doing anyway :).

Reply:
The order of items in the dictionary depends on how items are added, which means it's left up to the developer writing that plugin. If the algorithm only works if the items are in a certain order, it's best to sort them at the point you're using the data rather than relying on us to keep the order straight when creating the dictionary.

Reply:
+1 -- I've added this.